lib/iris/_concatenate.py

```diff
@@ -402,6 +413,14 @@ def _coordinate_differences(self, other, attr):
                       ', '.join(diff_names))
         return result
 
+    def promote_defn(self):
+        defn = self.defn
+        kwargs = defn._asdict()
+        if kwargs['dtype'] is None and self.data_type.kind == 'i':
+            kwargs['dtype'] = self.data_type
+            defn = iris.cube.CubeMetadata(**kwargs)
+        return defn
```

Comment:
Actually, I'm not going to do that here as this might all change given conversations with @pp-mo